crypto/provider_core.c

```diff
@@ -432,9 +453,45 @@ static int provider_activate(OSSL_PROVIDER *prov)
             prov->query_operation =
                 OSSL_get_provider_query_operation(provider_dispatch);
             break;
+#ifndef OPENSSL_NO_ERR
+# ifndef FIPS_MODE
+        case OSSL_FUNC_PROVIDER_GET_REASON_STRINGS:
+            p_get_reason_strings =
+                OSSL_get_provider_get_reason_strings(provider_dispatch);
+            break;
+# endif
+#endif
         }
     }
 
+#ifndef OPENSSL_NO_ERR
+# ifndef FIPS_MODE
+    if (p_get_reason_strings != NULL) {
+        const ERR_STRING_DATA *reasonstrings =
+            p_get_reason_strings(prov->provctx);
+        size_t cnt;
+
+        /*
+         * Because ERR_load_strings() patches the array's error number
+         * with the error library number, we need to make a copy of that
+         * array.
+         */
+        cnt = 1;                 /* One for the terminating item */
+        while (reasonstrings[cnt].error != 0)
+            cnt++;
+        prov->error_strings = OPENSSL_zalloc(sizeof(ERR_STRING_DATA) * cnt);
+        if (prov->error_strings == NULL)
+            return 0;
+        while (cnt-- > 0) {
```

Comment:
size_t can be unsigned so test ```!= 0``` seems more clear.

Reply:
I disagree specifically for this form.  It expresses "as long as the pre-decrement value is above zero", which is perfectly all right for unsigned values.